### ISSUE TITLE
native apple silicon ("M1" / aarch64) support

### ISSUE THREAD

**Issue by ThomasWaldmann** [cloudradar-monitoring/rport#280]
Just noticed that github releases only offers intel / x64 binaries.

BTW, thanks for the FrosCon talk!

**Comment by ThomasWaldmann** [comment]
From the docs:
```
On Mac OS (M1/Arm based)

Coming soon.
```

Seems like this is tracked in the docs / brains of the developers at least. ;-)

**Comment by thorstenkramm** [comment]
@ThomasWaldmann Yes. This is an open issue at the moment, and it will take a while until we provide binaries for M1.
Extending the build pipeline is not very complex, but it takes time and from other projects we have learned, that a Mac is used only for the first test and for playing around. Because our resources are limited, we focus on the Windows and Linux because these are the business relevant platforms. 
May I ask what's your planned use case on Mac?

**Comment by ThomasWaldmann** [comment]
Well, the M1 Macs are in the best selling laptops of 2020/2021, so guess some people use them (including me :).

Of course usually as client, not as servers, but easy ad-hoc maintenance behind NAT routers is a scenario which is rather frequent for client machines. This is why teamviewer/anydesk/etc. are so popular. Sadly, no popular/easy FOSS solutions yet.

rport maybe could be one, if it is easy enough and works on the platforms people use.

"easy to get running for the unprepared" is the killer criteria here. if you ever had some computer illiterate person at the phone, trying to download and run teamviewer or anydesk, needing half an hour for that and almost giving up, you know what I mean.

On my personal Macbook Air 2020 M1, I won't need remote maintenance as I usually have it with me.
But for other peoples' Macs, having a FOSS solution would be great, so one does not need proprietary tools to help them.

**Comment by philippdieter** [comment]
May I ask if there are any updates on this topic? I have the first case now where I would need the rport client on a M1 Mac. I just would like to know if I need to search for an alternative solution for this case.

**Comment by keglovitsadam** [comment]
+1 for M1 rport support.

**Comment by thorstenkramm** [comment]
@keglovitsadam Are you referring to a server or a client binary?
The code signing requirements are very time consuming and we need to rent an apple machine on the cloud. That creates costs. 
What is your use case? We still doubt there is a real market need.

**Comment by keglovitsadam** [comment]
@thorstenkramm I refer to client side. We would like to monitor / manage M1 Mac machines as well with rport.

**Comment by thorstenkramm** [comment]
@keglovitsadam How many M1 Macs do you want to manage?
Is it in a companies' environment? Or is it some kind of home lab setup?

**Comment by thorstenkramm** [comment]
Compiling the rport client on any Mac is straightforward.
```bash
VERSION=0.6.0
cd $(mktemp -d)
curl -LOSf https://github.com/cloudradar-monitoring/rport/archive/refs/tags/${VERSION}.tar.gz
tar xzf ${VERSION}.tar.gz
cd rport-${VERSION}
go build -v -ldflags="-X 'github.com/cloudradar-monitoring/rport/share.BuildVersion=${VERSION}'"  -o ./ ./cmd/rport/...
./rport --version
```

**Comment by keglovitsadam** [comment]
@thorstenkramm, thank you, I will try to compile the client. For now, its just for testing purposes for few Macs.

**Comment by keglovitsadam** [comment]
@thorstenkramm , I could compile and start on M1 Mac, it connects to the server but teher is notchin I can do. I mean I see some basic info about the machine but cannot run any simple script... Should I set something in the conf file other than the basic server and auth details? Thank you!

**Comment by thorstenkramm** [comment]
@keglovitsadam 
I recommend using the [default client configuration file](https://github.com/cloudradar-monitoring/rport/blob/master/rport.example.conf) with all options documented.
This allows you to properly set up command and [script execution](https://github.com/cloudradar-monitoring/rport/blob/0.6.0/rport.example.conf#L186).
